### DISCUSSION TITLE
the duplicate rr and missing end..attached as jpg

### DISCUSSION THREAD

**Discussion by MaheshNaikM** [graphics32/graphics32#264] [General] [answered]
i got the package generation errors again and i screen captured in .jpg format.
![fot  github](https://github.com/graphics32/graphics32/assets/5690156/b405c1cf-35db-4a7e-8b16-cc6f6d0f5262)
![github2](https://github.com/graphics32/graphics32/assets/5690156/de351d76-2713-4594-8c2b-fd87dd8090d2)
When i edit thw 3 errors and recompile i get thru. I noticed some files have incomplete line ends!

**Comment by andersmelander** [comment, marked as answer]
I believe the problem is at your end.
This is what the file (`Source\Packages\D104\GR32_D.dpk`) looks like when I pull it:
![billede](https://github.com/graphics32/graphics32/assets/18141921/483247a2-8e06-4b54-b24b-fba9153301b9)

You need to make sure that your Git client is configured to use CR/LF line endings:
https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings

I.e. execute: `git config --global core.autocrlf true`

**Reply by MaheshNaikM** [reply]
oh dear,my file also does not have the duplicates and has all lines properly ended..?...so the problem might be with embarcaderos delphi! . 
I  use the distribution to create a cbuilder usable package. [ I have had nasty problems some time back in getting  hpp, bpl files from pas files].
  the R package goes thru always,only the D package had problem sometimes--but not now?
   Thank you for the help
